### PR TITLE
[Fix] Zero values unavailable in DonutChart plot

### LINKED ISSUE #30051: [Bug]: [Donut chart] Hovering over a donut legend with 0 value does not put 0 in the centered value
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 11 10.0.22635
    CPU: (12) x64 Intel(R) Xeon(R) CPU E5-1650 v3 @ 3.50GHz
    Memory: 10.41 GB / 31.92 GB
  Browsers:
    Edge: Chromium (120.0.2210.61)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codepen.io/adriblt/pen/GRzaryb

## Actual Behavior
In a donut chart, hovering on a section or the legend updates the center text to have the associated value.
This does not work as expected for sections with a 0 value, instead it displays the total.

Default view:
image

Hovering legend with non zero velu:
image

Hovering legend with a 0 value:
image

## Expected Behavior
The centered value should be "0" when hovering the legend

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

https://make.powerautomate.com/

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by Shubhabrata08**
## Previous Behavior
Zero values didn't appear on legend hover inside donut
image

## New Behavior
Zero values now appear on hovering over legend with data zero. This was primarily due to filtering of data by >0 condition.
image

## Related Issue(s)

- Fixes #30051

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/a493617d5c710d78246d315e67fa7eb2d7b6b9f9'>a493617d5c710d78246d315e67fa7eb2d7b6b9f9</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/components/DonutChart/DonutChart.base.tsx [hunk 1]
```diff
@@ -138,7 +138,7 @@ export class DonutChartBase extends React.Component<IDonutChartProps, IDonutChar
                 focusedArcId={this.state.focusedArcId || ''}
                 href={this.props.href!}
                 calloutId={this._calloutId}
-                valueInsideDonut={this._toLocaleString(valueInsideDonut)}
+                valueInsideDonut={this._toLocaleString(valueInsideDonut?.toString())}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/DonutChart/DonutChart.base.tsx:141]: >toString [](http://example.com/codeflow?start=73&length=8)

why is this change needed. The tostring conversion is supposed to be performed by the toLocaleString function only.

Comment (Shubhabrata08) [reply packages/react-charting/src/components/DonutChart/DonutChart.base.tsx]: Reverted this change. Introduced fix in utilities.ts.

File: packages/react-charting/src/utilities/utilities.ts [hunk 2]
```diff
@@ -1351,7 +1351,7 @@ export const getAccessibleDataObject = (
 
 type LocaleStringDataProps = number | string | Date | undefined;
 export const convertToLocaleString = (data: LocaleStringDataProps, culture?: string): LocaleStringDataProps => {
-  if (!data) {
+  if (data === undefined || data === null || Number.isNaN(data)) {
```
Comment (AtishayMsft) [inline packages/react-charting/src/utilities/utilities.ts:1354]: Can you add a unit test for these cases - data = false, data = empty string and data = 0

Comment (Shubhabrata08) [reply packages/react-charting/src/utilities/utilities.ts:1354]: Ok

File: packages/react-charting/src/utilities/utilities.ts [hunk 3]
```diff
@@ -1351,7 +1351,7 @@ export const getAccessibleDataObject = (
...
     return data;
```
Comment (AtishayMsft) [inline packages/react-charting/src/utilities/utilities.ts:1355]: does isNan return true for string data also

Comment (Shubhabrata08) [reply packages/react-charting/src/utilities/utilities.ts:1355]: Returns False for data types other than Number

File: packages/react-charting/src/utilities/utilities.ts (hunk 2 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/utilities/utilities.ts:1354]: Couldn't add test for null as the function doesn't accept null as an input

Comment (Shubhabrata08) [reply packages/react-charting/src/utilities/utilities.ts:1354]: image
@AtishayMsft

Comment (AtishayMsft) [reply packages/react-charting/src/utilities/utilities.ts:1354]: can you add UTss for remaining cases. 
data = false/False
data = "Hello world"
data = "123"
data = ""
data = " "
data = null. You can set the variable being passed as null.

Comment (Shubhabrata08) [reply packages/react-charting/src/utilities/utilities.ts:1354]: Ok

Comment (Shubhabrata08) [reply packages/react-charting/src/utilities/utilities.ts:1354]: image

Setting the variable null isn't allowed by type LocaleStringDataProps 

image

**Comment by Gabnight** [comment]
Hello,

We've a bug on our product related to this fix.
Have you got any ETA for this fix to be merged, please?

File: packages/react-charting/src/utilities/UtilityUnitTests.test.ts [hunk 4]
```diff
@@ -17,7 +17,27 @@ describe('Unit test to convert data to localized string', () => {
...
-
+  test('Should return NaN when data is NaN', () => {
+    expect(utils.convertToLocaleString(NaN)).toBeNaN();
+  });
+  test('Should return localized 0 when data is numeric 0', () => {
+    expect(utils.convertToLocaleString(0)).toBe('0');
+  });
+  test('Should return localized 123 when data is string 123', () => {
+    expect(utils.convertToLocaleString('123')).toBe('123');
+  });
```
Comment (AtishayMsft) [inline packages/react-charting/src/utilities/UtilityUnitTests.test.ts:28]: the sentences are not matching the return values for some tests

Comment (Shubhabrata08) [reply packages/react-charting/src/utilities/UtilityUnitTests.test.ts:28]: Fixed statements

**Comment by Shubhabrata08** [comment]
@Gabnight The fix has been merged.

**Comment by Gabnight** [comment]
Thanks
